Sources/ViewPosition.swift

```diff
@@ -0,0 +1,155 @@
+    // MARK: Initialization
+    
+    public init(view: UIView, anchor: Anchor) {
+        self = ViewPosition(view: view, position: anchor.anchorPoint(onRect: view.bounds))
+    }
+    
+    public init(label: UILabel, anchor: Anchor) {
+        self = ViewPosition(view: label, position: anchor.anchorPoint(onRect: FontMetrics(for: label.font).textFrame(within: label.bounds)))
+    }
+    
+    public init(view: UIView, position: CGPoint) {
+        self.view = view
+        self.anchor = PixelRounder(for: view).roundToPixel(position)
+        originalBounds = view.bounds
+    }
+    
+    // MARK: Public Methods
+    
+    /// Aligns the receiver's view to the passed in ViewPosition.
+    public func align(to otherViewPosition: ViewPosition, xOffset: CGFloat = 0.0, yOffset: CGFloat = 0.0) {
+        guard view.bounds == originalBounds else {
+            assertionFailure("Bounds of view to align have changed since ViewPosition was created! \(originalBounds) -> \(view.bounds)")
+            return
+        }
+        guard otherViewPosition.view.bounds == otherViewPosition.originalBounds else {
+            assertionFailure("Bounds of view to align to have changed since ViewPosition was created! \(otherViewPosition.originalBounds) -> \(otherViewPosition.view.bounds)")
+            return
+        }
+        guard let superview = view.superview else {
+            assertionFailure("Attempting to align view but no superview exists! \(view)")
+            return
+        }
+        guard let otherSuperview = otherViewPosition.view.superview else {
+            assertionFailure("Attempting to align view but no superview exists! \(otherViewPosition.view)")
+            return
+        }
+        
+        let frameOriginToAlignTo = otherSuperview.convert(otherViewPosition.view.frame.origin, to: superview)
+        view.frame.origin = PixelRounder(for: view).roundToPixel(
+            CGPoint(x: frameOriginToAlignTo.x + otherViewPosition.anchor.x - anchor.x + xOffset,
+                    y: frameOriginToAlignTo.y + otherViewPosition.anchor.y - anchor.y + yOffset)
+        )
+    }
+    
+    /// Convenience to align the receiver's view to the superview's anchor.
+    public func align(toSuperviewAnchor superviewAnchor: Anchor, xOffset: CGFloat = 0.0, yOffset: CGFloat = 0.0) {
+        guard let superview = view.superview else {
+            assertionFailure("Trying to align to \(view)'s superview but no superview exists")
+            return
+        }
+        
+        if let superview = superview as? UILabel {
+            align(to: ViewPosition(label: superview, anchor: superviewAnchor), xOffset: xOffset, yOffset: yOffset)
+        } else {
+            align(to: ViewPosition(view: superview, anchor: superviewAnchor), xOffset: xOffset, yOffset: yOffset)
+        }
+    }
+    
+    // MARK: Internal Properties
+    
+    /// The pixel-rounded anchor point for positioning, in the coordinate space of the bounds.
+    internal let anchor: CGPoint
```

Comment:
Good q!

The `anchor` here is more flexible than that. `Anchor` defines a pre-defined position on a rect. The `ViewPosition`‘s `anchor` is meant to define an arbitrary position relative to the rect. Representing it as a `CGPoint` allows us to add/subtract offsets from the `ViewPosition`. Let me call this `anchorPoint` rather than `anchor` to avoid the confusion.